submissions/IhorOzerov/OOP exercise/index.js

```diff
@@ -0,0 +1,85 @@
+class Inhabitant {
+   constructor(creature, name, gender, legs, friends, say) {
+      this.creature = creature
+      this.name = name
+      this.gender = gender
+      this.legs = legs
+      this.friends = friends
+      this.say = say
+   }
+   inhabitantSpeech(creatureText, nameText, genderText, legsText, friendsText, sayText) { 
+      return `${creatureText}${this.creature}` + `${nameText}${this.name}` +
+         `${genderText}${this.gender}` + `${legsText}${this.legs}` +
+         `${friendsText}${this.friends}` + `${sayText}${this.say}`
+   }
+}
+
+class Animal extends Inhabitant{
+   constructor(creature, name, gender, friends, say) {
+      super(creature, name, gender, 4, friends, say);
+   }
+   inhabitantSpeech() {
+      return super.inhabitantSpeech(`Sup, i am a `, `, my name is `, `, i am cool `, `. With all of my `, ` legs and my friends, `,
+         ` I want `) 
+   }
+}
+
+class Cat extends Animal{
+   constructor(name, gender, friends) {
+      super("Cat", name, gender, friends, "to say Where is my money, Gregory?!");
+   }
+}
+
+class Dog extends Animal{
+   constructor(name, gender, friends) {
+      super("Dog", name, gender, friends, "to say that you are breathtaking!");
+   }
+}
+
+class Human extends Inhabitant{
+   constructor(name, gender, friends, say) {
+      super("Human", name, gender, 2, friends, say);
+      this.hands = 2
+   }
+   inhabitantSpeech() {
+      return super.inhabitantSpeech(`Hi, i am a `, `, my name is `, `, i am nice `, `. With my ` + this.hands + ` hands, `,
```

Comment:
As a peer developer tasked to work on this project I am quite confused with how hands are treated.
Looks like a limp insertion...
I am also designing a child class for birds and trying to re-use what's already done here.
So I put `super.inhabitantSpeech(` and IDE Intellisence tells me that the method expects to receive `creatureText, nameText, genderText, legsText, friendsText, sayText`, i.e. no place for hands or wings...
Should I follow the same coding pattern?

OOP can help you writing a better reading and consistent code.

Reply:
1. In the case of a bird, I think that hands or wings will be added in the same way because I can't give property "wings" to all inhabitants 
2. Please give some advice on what exactly I need to improve on that situation. [In my previous version](https://github.com/kottans/frontend-2022-homeworks/pull/515/commits/23fc8caf213dd1a269f3ed863053e679eac350f4) I had another way to make method, and this method looks better, but it doesn't contain properties in father class. If it's ok i will remove my code to previous version.